File: BAKEND/python/codigo-Python/algoritmos/selectionSort.py

def selectionSort(arreglo):
    arreglo_ord = arreglo.copy()
    # Iteracion sobre el arreglo para poder toma la posicion que va a tener el dato minimo segun se vaya encontrando 
    for i in range(0,len(arreglo_ord)-1):
        # Guardamos la posicion de i que simpre seria la posicion minima 
        pos_minino = i
        # Guardamos el numero que se encuentre en la iteracion de la posicion minima actual 
        minimo = arreglo_ord[i]
        # Iteracdor para poder buscar cual es el valor minimo en lo que resta del arreglo
        for j in range(pos_minino,len(arreglo_ord)):
            # Sentencia para encontar el dato que es menor a la posicon que estamos comparando que seria la minima
            if arreglo_ord[j]<minimo:
                # si es menor al que se encuentra en la posiicion de minimo se actualiza la posicion de minimo y el valor de mininmo
                minimo = arreglo_ord[j]
                pos_minino = j
        # si es que la poscision de minimo cambio, intercambiamos los datos de posicion 
        if pos_minino != i:
            # 
            arreglo_ord[i],arreglo_ord[pos_minino] = arreglo_ord[pos_minino],arreglo_ord[i]
    return arreglo_ord


# print(selectionSort([10,2,5,78,3,7,27,89,6]))

""" 
EL trabajo del algoritmo es el next 
    10,2,5,78,3,7,27,89,6
en la primera iteracion guarda la posicion de 0 , la ultima la ignora porque ps el ultimo ya quedaria por default cuando se recorre todo 
y tambien gurda el dato de esta posicion 
    10 y 0 
el dato y la posicion que en este momento gurarda el dato mas pequeño 
itera desde la posicion que es hasta el momento la que debe de tener el dato mas pequeño hasta el final de arreglo
si la posicion en j es menor a minimo cambia a la posicion deonde estaria el dato menor 
    2<10 si
        minimo = 2
        pos_minimo  = 1
si no es menor sigue iterando hasta que encuentre un valor menor  hasta que en cuentre uno y pude cambiar el valor de la posicion con el numeor mas pequeño , para que si el valor de posicion de la posicion minimo es diferente a la iteracion de i entonces se intercambien los valores 
    1 !=  0 si 
        arrreglo[0] = arreglo[pos_minimo] osea 2 
        arreglo[posminomo] osea 1 = arreglo[i] osea 10 
y despues de este cambio se itera sobre i y  ya actualizada la posicion minima y el valor minimo 
se itera sobre j para ewncontar el valor siguiente minimo 

EN CONCLUSION CREO QUE 
Se guarda un lugar se busca el valor que sea menor a los demas , depues de inetrecambien los valores de posicion , despues se toma la siguiente posicion para poner el segund valor menor en el arreglo que se haya encontrado y asi hasta llegar a la penultima porque el valor ultimo se obtiene automagicamente cuando se obtien todos los anteriores
"""

File: FRONTEND/javaScript/codigo-Js/edteam/algoritmos/selectionSort.js
function selectionSort(arreglo){
    arreglo_ord = arreglo.slice()
    for(i=0;i<arreglo_ord.length-1;i++){
        // se va a modificar cuando se encuentre un nuevo minimo en el ciclo de j pero cuando se salgasa y se vuelva a iterar en ciclo de i se va a setear con el siguiente valor de i que es la siguiente posicion de donde se coloco el valor minimo encontrado
        pos_delminimoActual = i
        valor_minimo = arreglo_ord[i]
        for(j=pos_delminimoActual;j<arreglo_ord.length;j++){
            if (arreglo_ord[j]<valor_minimo){
                valor_minimo = arreglo_ord[j]
                pos_delminimoActual = j
            }
        }
        if(pos_delminimoActual != i){
            aux = arreglo_ord[i]
            arreglo_ord[i] = arreglo_ord[pos_delminimoActual]
            arreglo_ord[pos_delminimoActual] = aux
        }
    }
    return arreglo_ord
}

console.log(selectionSort([2,4,67,6,5,7,1]))

/* 
Denuevo
La iteracion de i recorre la posicion donde se deve de guragar el nuevo valor minimo encintrado en todo el arreglo,desde la posicion que tenga el valor minimo hasta el final.

La iteracion en j es para poder encontrar el nuevo valorminimo y su posicion de este, despues si es que se encontro un valor que es menor al de la ultimo minimo se debe intercambir, 

Guardando el valor del arrglo en la posion de i en un auxiliaar para despues pasar el valor de este a la posicion donde se encontraba el nuevo minimo 
*/